builder/builder-context.c

```diff
@@ -108,6 +109,9 @@ builder_context_finalize (GObject *object)
   g_strfreev (self->cleanup_platform);
   glnx_release_lock_file(&self->rofiles_file_lock);
 
+  if (self->sources_dirs)
+    g_clear_pointer (&self->sources_dirs, g_ptr_array_unref);
```

Comment:
You don't need to check for `self->sources_dirs` being non-NULL with `g_clear_pointer()`